### PR TITLE
NEW: Staccato::Adapter::NetHttpViaProxy

### PR DESCRIPTION
**Pull request by snow**

### CODE REVIEW

File: lib/staccato/adapter/net_http_via_proxy.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    class NetHttpViaProxy
+
+      def initialize(uri, proxy_host, proxy_port)
+        @uri = uri
+        @proxy_host = proxy_host
+        @proxy_port = proxy_port
+      end
+
+      def post(params)
+        ::Net::HTTP.Proxy(@proxy_host, @proxy_port).post_form @uri, params
```
Comment (tpitale) [inline lib/staccato/adapter/net_http_via_proxy.rb]: Should this be ::Proxy?

Comment (snow) [inline lib/staccato/adapter/net_http_via_proxy.rb]: well, it's `.Proxy`, see http://ruby-doc.org/stdlib-2.3.1/libdoc/net/http/rdoc/Net/HTTP.html#method-c-Proxy

I've use this class in my project for months. After you suggest just use a separate adaptor for proxy, I just copied it here and removed keyword arguments.

**Comment by tpitale** [comment]
I'm speaking stylistically. Maybe I should add rubocop.

Class methods can be called with either a . or ::, but when the class method is made to appear like a constant, as in the case of "casting", you will mostly see folks use the :: style to further indicate that "casting" is happening. It's like pretending that the method is actually a constant.

**Comment by snow** [comment]
thanks for the hint~
I will update it later.

**Comment by tpitale** [comment]
Thanks @snow.

**Comment by snow** [comment]
my pleasure 😸

**Pull request merged**